apis/workloads/v1alpha1/console_types.go

```diff
@@ -44,6 +44,12 @@ type ConsoleSpec struct {
 	// The command and arguments to execute. If not specified the command from
 	// the template specification will be used.
 	Command []string `json:"command,omitempty"`
+
+	// Enable TTY and STDIN on the underlying container, so clients can attach
+	// interactively. This should usually be set to true; however, in certain
+	// situations, enabling the TTY on a container in the console causes
+	// breakage - in Tekton steps, for example.
+	Interactive bool `json:"interactive,omitempty"`
```

Comment:
Since this is a breaking change the idea would be to cut a new major version, and release a new version of Anu that did things correctly (and then hustle people to upgrade). 

The reason for making it a positive flag was simply that it reads better - it's weird to have lots of inverted boolean conditions around the place IMO. But it could be the other way if you'd prefer it (or, indeed, a boolean pointer).

Reply:
> then hustle people to upgrade

Hmmm yeah, that's the painful part here - it's not going to break in a graceful way, people are still going to be able to create consoles with the old version, they'll just be broken or may violate some expectations like "I can create a detached console and then later attach to it", until they've upgraded.

Really the proper way to add this option, in its current form, would be to cut a new version of the CRDs. But we don't have a good story around that, and I wouldn't recommend trying to tackle it as part of this PR either.

> it's weird to have lots of inverted boolean conditions around the place

Yeah fair point here. I think it does depend upon how you look at this feature though - you can very easily see it as "Interactive is the default, non-interactive is an optional modifier" (but arguably that's an artefact of the project's history). We currently only check for this condition twice, and in one place we've already inverted the boolean, so it might not be that bad.

---

I'll defer the decision on this one to you and the team though - given that you'll be handling the coordination of the rollout you're probably best placed to the effort of managing users and tradeoffs of risks.

Reply:
I'm finding it really hard to separate maybe being biased because it's how they are already from actually thinking consoles should be interactive by default.

My gut says interactive by default, but I don't have that strong a reason beyond it matching what the wrapping CLI will request by default. Not sure if that's a good reason for the library to behave that way.

Reply:
I changed the flag in CreateOptions - this is in fact the interface that Anu uses so should allow a seamless upgrade.

Reply:
Maybe I'm missing something here, but I'm not sure it does provide a seamless upgrade?

People will currently have binaries that have no knowledge of the `Interactive` field on `ConsoleSpec`, and won't be setting it in their version of the `CreateResource` function. This means that we're going to be creating `Console` objects that are missing the `Interactive` field, so setting it to implicitly false

If we do go down this route then we need to make sure that all human users are upgraded before we roll out the new controller, as well as ensuring that any automation (PS DB migrations?) still works without a TTY present (some processes get funny and perform different behaviour when a TTY isn't found)

Reply:
... GODDAMMIT

You're right obviously. I've pushed a change but will need to deploy the controller to lab to test.